From 9dd2bef9ed7e14e19d613852e37be7f2e8ba381d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tuomas=20R=C3=A4s=C3=A4nen?= <tuomas.rasanen@opinsys.fi>
Date: Fri, 17 May 2024 18:04:39 +0300
Subject: [PATCH 09/14] [puavo] appearance: change theme existence check to
 test only the existence of theme dir

Not all themes have index.theme files, e.g KvArc.
---
 panels/background/cc-background-panel.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/panels/background/cc-background-panel.c b/panels/background/cc-background-panel.c
index 9814f88a0..844536b34 100644
--- a/panels/background/cc-background-panel.c
+++ b/panels/background/cc-background-panel.c
@@ -167,14 +167,14 @@ get_default_theme(const gchar *theme)
 static gboolean
 does_theme_exist(const gchar *theme_name, const gchar *theme_name_suffix)
 {
-  gchar *theme_index_file_path;
+  gchar *theme_path;
   gboolean retval = FALSE;
 
-  theme_index_file_path = g_strconcat ("/usr/share/themes/", theme_name, theme_name_suffix, "/index.theme", NULL);
-  if (g_access(theme_index_file_path, F_OK) == 0)
+  theme_path = g_strconcat ("/usr/share/themes/", theme_name, theme_name_suffix, NULL);
+  if (g_access(theme_path, F_OK) == 0)
     retval = TRUE;
 
-  g_free(theme_index_file_path);
+  g_free(theme_path);
 
   return retval;
 }
-- 
2.39.2

